packages/shell-api/src/shell-bson.ts

```diff
@@ -0,0 +1,71 @@
+/* eslint prefer-rest-params: 0, @typescript-eslint/ban-ts-ignore: 0 */
+import bson from 'bson';
+
+function dateHelper(...args: any[]): Date {
```

Comment:
```
type DateConstructorArguments = [any, any, ...any[]];

function dateHelper(...args: DateConstructorArguments): Date {
  if (args.length === 0) {
    return new Date();
  }
  if (args.length === 1) {
    return new Date(args[0]);
  }

  return new Date(Date.UTC(...args));
}
```

Reply:
Happy to make the change, but can you explain why that's better than just any[]? It is still failing the typescript compilation with that chage.
<img width="925" alt="Screen Shot 2020-05-05 at 14 12 35" src="https://user-images.githubusercontent.com/1045019/81064764-901c2600-8eda-11ea-9dd5-edb184d05940.png">

Reply:
that worked for me, let me pull again. `any[]` does not say anything about the cardinality of the array, which means it would match an empty array for example.

`[any, any, ...any[]]` says at least 2 elements has to be there.

Reply:
Neat trick!

Reply:
This works for me: https://github.com/mongodb-js/mongosh/blob/bson-fix-test-ts/packages/shell-api/src/shell-bson.ts